### PR TITLE
Basic CSharp packaging support for ROCm EP

### PR DESCRIPTION
**Pull request by cloudhan**
This PR mainly fixes building errors when trying to build nupkg for ROCm EP.
It also slighly improve the packaging logic so that devlopers can produce the nupkg on linux natively.

### CODE REVIEW

File: include/onnxruntime/core/session/onnxruntime_c_api.h [hunk 1]
```diff
@@ -4239,6 +4239,16 @@ struct OrtCustomOp {
...
 
+/*
+ * This is the old way to add the ROCm provider to the session, please use
+ * SessionOptionsAppendExecutionProvider_ROCM above to access the latest functionality
+ * This function always exists, but will only succeed if Onnxruntime was built with
+ * HIP support and the ROCm provider shared library exists
+ *
+ * \param device_id HIP device id, starts from zero.
+ */
+ORT_API_STATUS(OrtSessionOptionsAppendExecutionProvider_ROCM, _In_ OrtSessionOptions* options, int device_id);
```
[the remark this answers is not in the document]
Comment (cloudhan) [reply include/onnxruntime/core/session/onnxruntime_c_api.h:4250]: This is an old API, it belongs to the global scope, no struct involved, so there is no need for initializing as in `OrtApi` stuct. I am doubt this decl will affect the ABI tho. The API has already implemented at https://github.com/microsoft/onnxruntime/blob/9d26f8f4fe1654bfe7e2f631f892a8cf08caf050/onnxruntime/core/session/provider_bridge_ort.cc#L1557-L1562. This decl just expose it, so the C# dll can use it to initialize a sess with the `devcie_id` API. Subsequent PR adds API for initializing with provider options.

**Review by yuslepukhin — DISMISSED** [review]
:shipit:

File: include/onnxruntime/core/session/onnxruntime_c_api.h [hunk 2]
```diff
@@ -4280,6 +4280,16 @@ struct OrtCustomOp {
  */
 ORT_API_STATUS(OrtSessionOptionsAppendExecutionProvider_CUDA, _In_ OrtSessionOptions* options, int device_id);
 
+/*
+ * This is the old way to add the ROCm provider to the session, please use
```
Comment (yuslepukhin) [inline include/onnxruntime/core/session/onnxruntime_c_api.h:4284]: >This is the old way [](http://example.com/codeflow?start=3&length=19)

Should we expose the old way?

**Review by yuslepukhin — APPROVED** [review]
:shipit:

**Pull request merged**